runtime/tests/interpreter/interface_test.go

```diff
@@ -241,3 +242,425 @@ func TestInterpretInterfaceDefaultImplementationWhenOverriden(t *testing.T) {
+		require.NoError(t, err)
+		assert.Equal(t,
+			interpreter.NewUnmeteredIntValueFromInt64(18),
+			value,
+		)
+
+		// Implementation should satisfy inherited conditions
+		_, err = inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(5))
+		utils.RequireError(t, err)
+		assert.ErrorAs(t, err, &interpreter.ConditionError{})
+	})
+
+	t.Run("conditions in both", func(t *testing.T) {
+
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            struct interface A {
+                pub fun test(_ a: Int): Int {
+                    pre { a < 20 }
+                }
+            }
+
+            struct interface B: A {
+                pub fun test(_ a: Int): Int {
+                    pre { a > 10 }
+                }
+            }
+
+            struct C: B {
+                fun test(_ a: Int): Int {
+                    return a + 3
+                }
+            }
+
+            pub fun main(_ a: Int): Int {
+                let c = C()
+                return c.test(a)
+            }
+        `)
+
+		value, err := inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(15))
+		require.NoError(t, err)
+		assert.Equal(t,
+			interpreter.NewUnmeteredIntValueFromInt64(18),
+			value,
+		)
+
+		// Implementation should satisfy both inherited conditions
+
+		_, err = inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(5))
+		utils.RequireError(t, err)
+		assert.ErrorAs(t, err, &interpreter.ConditionError{})
+
+		_, err = inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(25))
+		utils.RequireError(t, err)
+		assert.ErrorAs(t, err, &interpreter.ConditionError{})
+	})
+
+	t.Run("conditions from two paths", func(t *testing.T) {
+
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            struct interface A {
+                pub fun test(_ a: Int): Int {
+                    pre { a < 20 }
+                }
+            }
+
+            struct interface B {
+                pub fun test(_ a: Int): Int {
+                    pre { a > 10 }
+                }
+            }
+
+            struct interface C: A, B {}
+
+            struct D: C {
+                fun test(_ a: Int): Int {
+                    return a + 3
+                }
+            }
+
+            pub fun main(_ a: Int): Int {
+                let d = D()
+                return d.test(a)
+            }
+        `)
+
+		value, err := inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(15))
+		require.NoError(t, err)
+		assert.Equal(t,
+			interpreter.NewUnmeteredIntValueFromInt64(18),
+			value,
+		)
+
+		// Implementation should satisfy both inherited conditions
+
+		_, err = inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(5))
+		utils.RequireError(t, err)
+		assert.ErrorAs(t, err, &interpreter.ConditionError{})
+
+		_, err = inter.Invoke("main", interpreter.NewUnmeteredIntValueFromInt64(25))
+		utils.RequireError(t, err)
+		assert.ErrorAs(t, err, &interpreter.ConditionError{})
+	})
```

Comment:
Re: https://github.com/onflow/flips/pull/40/files#r1178153289

In what order are pre/post conditions run if there is complex DAG of interfaces?

Reply:
Currently, it is a depth-first traversal. I'll add a test case for that 👍

Reply:
Added here: 9b184ac14bdbbb51f24eca9b09a4a1629c60b337

Reply:
One potential reason to delay this until Stable Cadence: until the `view` changes restrict mutations in conditions, this may present an attack vector because people will be able to execute arbitrary code in conditions and thus get around the prevention of overriding default implementations.

Reply:
I think it's possible even today, without inheritance. One can invoke arbitrary functions in conditions of two interfaces implemented by a concrete type, where having two default implementations is not allowed.

Interface inheritance in a way only 'flattens' a hierarchy, and is more of a desugar for explicitly writing them down in the concrete implementation.
e.g:
```cadence
struct Foo: A {}

struct interface A : B {}
struct interface B : C {}
struct interface C {}
```

has the same effect as:
```cadence
struct Foo: A, B, C {}

struct interface A {}
struct interface B {}
struct interface C {}
```